src/block-management/on-new-block.js

```diff
@@ -0,0 +1,19 @@
+"use strict";
+
+var assign = require("lodash.assign");
+var reprocessTransactions = require("../transact/reprocess-transactions");
+var isObject = require("../utils/is-object");
+
+function onNewBlock(block) {
+  return function (dispatch) {
+    if (isObject(block)) {
+      dispatch({
+        type: "SET_CURRENT_BLOCK",
+        block: assign({}, block, { number: parseInt(block.number, 16) })
```

Comment:
Since this PR will be a semver major change, consider taking the opportunity to no longer do this mutation on the block and instead store the raw block?  Having one field on the block be not in the native format can be confusing for users.  Alternatively, we could create a "ParsedBlock" that turns everything into byte arrays/BigNumbers so the stored block is easier to work with.